### ISSUE TITLE
Expansion breaks if $$ ends a line

### ISSUE THREAD

**Issue by igoralmeida** [brennier/vimtexer#8]
Hi,

Sometimes I have a line that will end with some `$math$`
(like the previous one, no trailing spaces). For lines above the problematic one, quicktex works fine and my mappings are expanded accordingly, but for lines after it the expansion does not work.

I took a quick look and I wonder if [this line](https://github.com/brennier/quicktex/blob/master/autoload/quicktex/mathmode.vim#L56) in `quicktex#mathmode#InMathMode` is the problem. Replacing the substitute command with `0,.-s/\\\@<!\$.\?//ne` (just added a `\?` at the end) works for me®, but this might have deeper implications.

Thoughts?

**Comment by brennier** [comment]
I'm surprised that this hasn't been reported yet. (I personally use `\( \)` instead of `$ $`.) Your suggested fix seems to work, so I've committed it. Thanks for catching this bug!

**Comment by igoralmeida** [comment]
It seems this is not entirely correct as I've experienced the problem again after opening the file. Same situation: the previous line ended with `$math$` and my expansions did not work until I added a character (in my case a trailing space) after the dollar sign.

I could "solve" it just by issuing `:source path/to/github.com/brennier/quicktex/autoload/quicktex/mathmode.vim` (and removing the trailing whitespace), which is weird because the script already had the `\?` bit from my previous attempt. Is there someplace else I would need to edit, perhaps?

I'll try to create a minimal example and submit in the next few (several?) days.
